lib/response.js

```diff
@@ -166,12 +165,14 @@ module.exports = {
     }
 
     // stream
-    if ('function' == typeof val.pipe) {
+    if (val instanceof Stream) {
       onFinish(this.res, destroy.bind(null, val));
-      ensureErrorHandler(val, err => this.ctx.onerror(err));
 
-      // overwriting
-      if (null != original && original != val) this.remove('Content-Length');
+      if (original != val) {
+        val.once('error', err => this.ctx.onerror(err));
```

Comment:
The two implementations seem to be not equal. We should only add `error` event listener when the stream has no other error event listener, which aim to avoid uncaught exception.

Also IMO koa's code philosophy is to keep code simple, so we don't treat 'one-liner module' as a bad thing.

Reply:
In current implementation the check in the `error-inject` does nothing (as to inline created functions never the same) and just always add `.on('error')` handler. Do you mean that was desired behaviour?
I'm proposing to remove `error-inject` here not because it's one line, but because even you (module author) forgot that you breaking it functionality while refactored to inline function. It just does nothing in current version.

Reply:
i don't understand what you mean, can you explain in a test?

Reply:
seems like a developer error

Reply:
It is, but the most probably explanation here that by hiding the actual logic in a one-line module even the author of the module forgot the logic on next refactoring and broke the functionality. That's what this PR is all about - fix the error and bring the logic inside to prevent such a problem in the future.